provision/3rd_party/Makefile.am

```diff
@@ -12,28 +12,52 @@ IPXE_DIR = ipxe-$(IPXE_VERSION)
 
 
 prep:
-	@ if [ ! -d "_work/$(IPXE_DIR)" ]; then \
-		echo "Extracting IPXE" ;\
-		mkdir -p _work/ ;\
-		tar -xJf $(IPXE_SOURCE) -C _work/ ;\
+	@ if [ -n "@local_ipxe_snp_i386_path@" -a -f "@local_ipxe_snp_i386_path@" -a \
```

Comment:
Would it be reasonable to have a single autoconfig variable that signifies local_ipxe or not local_ipxe instead of a variable for each binary, so we don't have so much code in the Makefile for this?

Also, I'm thinking if local_<app> is defined and the file doesn't exist we should bail out as an error instead of falling back to building the internal version. Otherwise we're going to be hiding a potential mismatch between what the configure script found and the current state of the system.